osu.Game.Rulesets.Osu/Objects/Drawables/DrawableOsuBlinds.cs

```diff
@@ -0,0 +1,298 @@
+        /// Percentage of playfield to extend blinds over. Basically moves the origin points where the blinds start.
+        /// </para>
+        /// <para>
+        /// -1 would mean the blinds always cover the whole screen no matter health.
+        /// 0 would mean the blinds will only ever be on the edge of the playfield on 0% health.
+        /// 1 would mean the blinds are fully outside the playfield on 50% health.
+        /// Infinity would mean the blinds are always outside the playfield except on 100% health.
+        /// </para>
+        /// </summary>
+        private const float leniency = 0.1f;
+
+        /// <summary>
+        /// Multiplier for adding a gap when the Easy mod is also currently applied.
+        /// </summary>
+        private const float easy_position_multiplier = 0.95f;
+
+        public DrawableOsuBlinds(Container restrictTo, bool hasEasy, bool hasHardrock)
+        {
+            this.restrictTo = restrictTo;
+
+            modEasy = hasEasy;
+            modHardrock = hasHardrock;
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(ISkinSource skin, TextureStore textures)
+        {
+            RelativeSizeAxes = Axes.Both;
+            Width = 1;
+            Height = 1;
+
+            Add(box1 = new Box
+            {
+                Anchor = Anchor.TopLeft,
+                Origin = Anchor.TopLeft,
+                Colour = Color4.Black,
+                RelativeSizeAxes = Axes.Y,
+                Width = 0,
+                Height = 1,
+                Depth = black_depth
+            });
+            Add(box2 = new Box
+            {
+                Anchor = Anchor.TopRight,
+                Origin = Anchor.TopRight,
+                Colour = Color4.Black,
+                RelativeSizeAxes = Axes.Y,
+                Width = 0,
+                Height = 1,
+                Depth = black_depth
+            });
+
+            Add(bgPanelLeft = new ModBlindsPanelSprite {
+                Origin = Anchor.TopRight,
+                Colour = Color4.Gray,
+                Depth = bg_panel_depth + 1
+            });
+            Add(panelLeft = new ModBlindsPanelSprite {
+                Origin = Anchor.TopRight,
+                Depth = bg_panel_depth
+            });
+
+            Add(bgPanelRight = new ModBlindsPanelSprite {
+                Origin = Anchor.TopLeft,
+                Colour = Color4.Gray,
+                Depth = fg_panel_depth + 1
+            });
+            Add(panelRight = new ModBlindsPanelSprite {
+                Origin = Anchor.TopLeft,
+                Depth = fg_panel_depth
+            });
+
+
+            random = new Random();
+            Add(bgRandomNpc = new Box
+            {
+                Anchor = Anchor.Centre,
+                Origin = Anchor.Centre,
+                Colour = Color4.Black,
+                Width = 512 * 0.4f,
+                Height = 512 * 0.95f,
+                RelativePositionAxes = Axes.Y,
+                X = -512,
```

Comment:
The randomNPC should appear and exit offscreen offscreen

Reply:
that's not possible because if you make the window really wide it will leap from the right to the left with extreme speed and not look very good. Right now it fades in to the right of the playfield and depending on the skin also to the right of the blind panel and then floats to the other side and fades out again